runtime/runtime/src/ext.rs

```diff
@@ -228,30 +228,35 @@ impl<'a> External for RuntimeExt<'a> {
         &mut self,
         data_id: CryptoHash,
         data: Vec<u8>,
-    ) -> Result<(), VMLogicError> {
+    ) -> Result<bool, VMLogicError> {
         // If the yielded promise was created by a previous transaction, we'll find it in the trie
         if let Some(yielded_promise) =
             get_yielded_promise(self.trie_update, data_id).map_err(wrap_storage_error)?
         {
             // Yields are only resumable by the account which created them
-            if yielded_promise.account_id != *self.account_id {
+            if &yielded_promise.account_id != self.account_id {
+                // NB: This is a contract programming error, so we want to return
                 return Err(
                     HostError::YieldedPromiseNotFound { data_id: data_id.to_string() }.into()
                 );
```

Comment:
Question here: do we want to return an error here or should we "simply" handle these as-if there wasn't a promise in the first place?

Reply:
Pros for an error: users get told they are doing something definitely wrong that ain't gonna wrong;
Pros for `Ok(false)`: it does not expose unnecessary state from the runtime to the contracts. Maintaining compatibility here might be difficult in the future...